nodejs/eks/examples/migrate-nodegroups/nginx-ing-cntlr.ts

```diff
@@ -0,0 +1,246 @@
+import * as k8s from "@pulumi/kubernetes";
+import * as input from "@pulumi/kubernetes/types/input";
+import * as pulumi from "@pulumi/pulumi";
+import * as rbac from "./nginx-ing-cntlr-rbac";
+
+// Create the NGINX Ingress Controller Service, RBAC, Configmap, Deployment,
+// and Pod Disruption Budget.
+export function create(
+    name: string,
+    labels: pulumi.Input<any>,
+    provider: k8s.Provider,
+    replicas: pulumi.Input<number>,
+    namespace: pulumi.Input<string>,
+    ingressClass: pulumi.Input<string>,
+    affinity: input.core.v1.Affinity,
+    tolerations: input.core.v1.Toleration[],
+): k8s.apps.v1.Deployment {
+
+    const defaultHttpBackendName = "nginx-default-http-backend";
+
+    // ServiceAccount.
+    const serviceAccount = rbac.makeNginxServiceAccount(
+        name, provider, namespace);
+    const serviceAccountName = serviceAccount.metadata.apply(m => m.name);
+
+    // RBAC ClusterRole.
+    const clusterRole = rbac.makeNginxClusterRole(
+        name, provider);
+    const clusterRoleName = clusterRole.metadata.apply(m => m.name);
+    const clusterRoleBinding = rbac.makeNginxClusterRoleBinding(
+        name, provider, namespace,
+        serviceAccountName, clusterRoleName);
+
+    // RBAC Role.
+    const role = rbac.makeNginxRole(
+        name, provider, namespace, ingressClass);
+    const roleName = role.metadata.apply(m => m.name);
+    const roleBinding = rbac.makeNginxRoleBinding(
+        name, provider, namespace,
+        serviceAccountName, roleName);
+
+    // NGINX Settings ConfigMap.
+    const configMap = makeConfigMap(name, provider, labels, namespace);
+    const configMapName = configMap.metadata.apply(m => m.name);
+
+    // Assemble the resources.
+    // Per: https://itnext.io/kubernetes-ingress-controllers-how-to-choose-the-right-one-part-1-41d3554978d2
+    const resources = {
+        requests: {memory: "1024Mi"},
+        limits: {memory: "2048Mi"},
+    };
+
+    // Create the Deployment.
+    const deployment = makeDeployment(name, provider,
```

Comment:
Would be nice to use named params here or create an input type.

Reply:
Done